var/spack/repos/builtin/packages/py-pyprecice/setup-v2.0.0.1.patch

```diff
@@ -0,0 +1,55 @@
+diff --git a/setup.py b/setup.py
+index 8c554a1..ef7c221 100644
+--- a/setup.py
++++ b/setup.py
+@@ -1,7 +1,17 @@
+ import os
+ import subprocess
+-from enum import Enum
++import warnings
++from packaging import version
++import pip
++
++if version.parse(pip.__version__) < version.parse("19.0"):
++    # version 19.0 is required, since we are using pyproject.toml for definition of build-time depdendencies. See https://pip.pypa.io/en/stable/news/#id209
++    warnings.warn("You are using pip version {}. However, pip version > 19.0 is recommended. You can continue with the installation, but installation problems can occour. Please refer to https://github.com/precice/python-bindings#build-time-dependencies-cython-numpy-defined-in-pyprojecttoml-are-not-installed-automatically for help.".format(pip.__version__))
++
++if version.parse(pip.__version__) < version.parse("10.0.1"):        
++    warnings.warn("You are using pip version {}. However, pip version > 10.0.1 is required. If you continue with installation it is likely that you will face an error. See https://github.com/precice/python-bindings#version-of-pip3-is-too-old".format(pip.__version__))
```

Comment:
Is this patch backwards? Don't we want to remove this pip logic? Since it's only a warning we don't really need to, we can just ignore it.

Reply:
Yes, this patch is indeed backwards. My idea/rationale was the following:
- `setup-v2.0.0.1.patch` ensures that the installation routine for all versions `2.0.0.1:2.1.1.99` is the same. This is easy since only `2.0.0.1` is affected by this.
-  `deactivate-version-check-via-pip.patch` (or its successor patch) will fix all the installation issues for `2.0.0.1:2.1.1.99`.

I hope to keep the patch complexities low with this approach as both patches have a clear intent and one only needs to fix the installation routine of the most common `setup.py`. In case we would need change to the installation procedure later again, we would only need to fix the second patch.  If it is advised against this, I have no problem changing this approach.